### ISSUE TITLE
Confusing error message in elm reactor when missing main function

### ISSUE THREAD

**Issue by brainrape** [elm/compiler#2093]
**Quick Summary:** 
When loading a module without a main function from elm reactor, the following error message is displayed:

```
Initialization Error

ReferenceError: _Platform_export is not defined
```

A better message would include a hint, eg.
> Did you forget to define a `main` function?

## SSCCE

```elm
module Main exposing (..)

a = 0
```

- **Elm:** 0.19.1
- **Browser:** *
- **Operating System:** *

**Comment by jessbowers** [comment]
This makes elm reactor confusing for beginners. That's what i'm experiencing now. There isn't easy documentation about getting started with reactor to begin with -- it's a little too simple in the guide. I don't know what to expect when pointing at my "non-main" elm module file. So this change would help that, IMO.

**Comment by nilslindemann** [comment]
Actually I like that oneliner in the docs: "use `elm reactor`". How could it be simpler? And what would you need more? "For more info use `elm --help`"?

**Comment by HenryFBP** [comment]
Just found this thread, thank god - Learning Helm, this is extremely confusing - Perhaps a warning can be thrown sometime during compilation?

**Comment by ssfrr** [comment]
Agreed this is a speedbump in an otherwise smooth onboarding experience. My sequence trying out Elm was

1. `brew install elm`
2. Run `elm` to see what it does, it looks like `elm init` is what I want
3. run `elm init`, which points me to [this doc](https://github.com/elm/compiler/blob/master/hints/init.md)
4. Based on that doc, I create a blank `Main.elm` run `elm reactor`, and double-click my Main.elm file
5. The first error I get says the compiler wants a declaration like `greet name = ...`, so I add a declaration and reload
6. The second error says the compiler wants a Main module and gives an example, so I add that and reload
7. The browser displays "Initialization Error: ReferenceError: _Platform_export is not defined"
8. Google for the error message find this issue.

The level of hand-holding to get to this point was pretty great, and the guidance from the compiler was very helpful, which made it all the more jarring when this popped up.
